csharp/src/Ice/Object.cs

```diff
@@ -125,6 +125,22 @@ public abstract class ObjectImpl : Object
     /// <returns>The return value is always ::Ice::Object.</returns>
     public static string ice_staticId() => "::Ice::Object";
 
+    public static void iceCheckMode(OperationMode expected, OperationMode received)
+    {
+        Debug.Assert(expected is OperationMode.Normal or OperationMode.Idempotent); // We never expect Nonmutating
+        if (expected != received)
+        {
+            if (expected is OperationMode.Idempotent && received is not OperationMode.Normal)
```

Comment:
Avoid using the deprecated Nonmutating enumerator. Semantics doesn't change.

Reply:
We should also review this check (in all languages) and sync-up with the IceRPC logic in a follow-up PR.

received = Normal should never trigger an error.

Reply:
See #3679